app/main/views/suppliers.py

```diff
@@ -422,3 +377,103 @@ def register_interest_in_framework(supplier_id, framework_slug):
             return jsonify(message="Database Error: {0}".format(e)), 400
 
         return jsonify(frameworkInterest=interest_record.serialize()), 201
+
+
+@main.route('/suppliers/<supplier_id>/frameworks/<framework_slug>', methods=['PUT'])
+def register_framework_interest(supplier_id, framework_slug):
+    updater_json = validate_and_return_updater_request()
+
+    framework = Framework.query.filter(
+        Framework.slug == framework_slug
+    ).first_or_404()
+
+    supplier = Supplier.query.filter(
+        Supplier.supplier_id == supplier_id
+    ).first()
+
+    if supplier is None:
+        abort(404, "supplier_id '{}' not found".format(supplier_id))
+
+    interest_record = SupplierFramework.query.filter(
+        SupplierFramework.supplier_id == supplier_id,
+        SupplierFramework.framework_id == framework.id
+    ).first()
+
+    if interest_record:
```

Comment:
I think this behaviour might be surprising. There's no indication that the endpoint doesn't take a payload and only saves supplier and framework id, so while it makes sense to return 200 if we know the data is the same, none of the side effects are triggered (i.e. there's no audit event with update details about the request) and there's no indication that any additional data that could've been sent in the request was discarded.

I suggest we either check that request doesn't contain anything besides updater details and keep the 200 or we reject repeated PUT requests (they shouldn't happen anyway once the "Make an application" button is hidden on the supplier frontend)